pkg/v3/plugin/factory.go

```diff
@@ -111,3 +121,42 @@ func (factory *pluginFactory) NewReportingPlugin(c ocr3types.ReportingPluginConf
 
 	return p, info, nil
 }
+
+func sampleFromProbability(rounds, nodes int, probability float32) (sampleRatio, error) {
+	var ratio sampleRatio
+
+	if rounds <= 0 {
+		return ratio, fmt.Errorf("number of rounds must be greater than 0")
+	}
+
+	if nodes <= 0 {
+		return ratio, fmt.Errorf("number of nodes must be greater than 0")
+	}
+
+	if probability > 1 || probability <= 0 {
+		return ratio, fmt.Errorf("probability must be less than 1 and greater than 0")
+	}
+
+	r := complex(float64(rounds), 0)
+	n := complex(float64(nodes), 0)
+	p := complex(float64(probability), 0)
+
+	g := -1.0 * (p - 1.0)
```

Comment:
Could we add a comment here just to give a high level overview of what's happening below?